File: bench/python/04-classes-traits/03-value-class-update.py
class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def translated(self, dx, dy):
        return Point(self.x + dx, self.y + dy)

point = Point(1.0, 2.0).translated(3.0, 4.0)
print(f"{point.x:g}")
print(f"{point.y:g}")

File: bench/rust/04-classes-traits/03-value-class-update.rs
struct Point { x: f64, y: f64 }
impl Point { fn translated(&self, dx: f64, dy: f64) -> Point { Point { x: self.x + dx, y: self.y + dy } } }
fn main() { let next = Point { x: 1.0, y: 2.0 }.translated(3.0, 4.0); println!("{}\n{}", next.x, next.y); }
